airflow/utils/session.py

```diff
@@ -46,20 +46,23 @@ def provide_session(func: Callable[..., RT]) -> Callable[..., RT]:
     database transaction, you pass it to the function, if not this wrapper
     will create one and close it for you.
     """
+    func_params = func.__code__.co_varnames
```

Comment:
Won't this be still called everytime `provide_session` decorator is called?

Reply:
This will be called only where the decorator is used, not every time the _Function_ is called